src/videojs-contrib-hls.js

```diff
@@ -87,6 +87,9 @@ export default class HlsHandler extends Component {
     // HlsSourceHandler.handleSource()
     // @link https://github.com/videojs/video.js/pull/3245
     this.options_ = videojs.mergeOptions(options, videojs.options.hls || {});
```

Comment:
Looking over this, the order seems wrong. Object-level options should trump page-level and this looks like it does the opposite. Could that be what's making the code below hard?